File: iteration2/src/Lecturer.java
import java.util.ArrayList;
public class Lecturer extends Person{

    private ArrayList<Course> givenCourses;
    private String academicTitle;

    public Lecturer(String name, String surname, String email, String phoneNumber, String ID, String password, String faculty, String department, ArrayList<Course> givenCourses, String academicTitle ) {
        super(name, surname, email, phoneNumber, ID, password, faculty, department);
        this.givenCourses = givenCourses;
        this.academicTitle = academicTitle;
    }
    public Lecturer(String name, String surname, String email, String phoneNumber, String ID, String password, String faculty, String department, String academicTitle ) {
        super(name, surname, email, phoneNumber, ID, password, faculty, department);
        this.academicTitle = academicTitle;
    }

    public ArrayList<Course> getGivenCourses() {
        return givenCourses;
    }
    public void setGivenCourses(ArrayList<Course> givenCourses) {
        this.givenCourses = givenCourses;
    }
    public String getAcademicTitle() {
        return academicTitle;
    }   
    public void setAcademicTitle(String academicTitle) {
        this.academicTitle = academicTitle;
    }
    public void setCourse(Course course){
        this.givenCourses.add(course);
    }
}

File: iteration3/src/lecturer.py
from person import Person
class Lecturer(Person):
    def __init__(self, name, surname, email, phone_number, ID, password, faculty, department, given_courses):
        super().__init__(name, surname, email, phone_number, ID, password, faculty, department)
        self.__given_courses = given_courses
        self.__academic_title = None


    def get_given_courses(self):
        return self.__given_courses
    
    def set_given_courses(self, given_courses):
        self.__given_courses = given_courses
    
    def get_academic_title(self):
        return self.__academic_title
    
    def set_academic_title(self, academic_title):
        self.__academic_title = academic_title
    
    def set_course(self, course):
        self.__given_courses.append(course)

